tests/test_products/test_product_routers.py

```diff
@@ -38,45 +38,57 @@ def test_authenticated_user_get_single_product(
     assert response.json()["id"] == db_products[1].id
 
 
-def test_authenticated_user_can_update_product(
+def test_superuser_can_update_product(
     sync_client: TestClient,
-    auth_headers: dict[str, str],
+    superuser_auth_headers: dict[str, str],
     db_products: list[ProductOutputSchema],
     db_categories: list[CategoryOutputSchema],
 ):
     update_data = ProductInputSchemaFactory.build(
         name="test_name", price=14.88, category_ids=[db_categories[0].id]
     )
     response = sync_client.patch(
-        f"products/{db_products[0].id}", data=update_data.json(), headers=auth_headers
+        f"products/{db_products[0].id}",
+        data=update_data.json(),
+        headers=superuser_auth_headers,
     )
 
     assert response.status_code == status.HTTP_200_OK
     assert response.json()["name"] == update_data.name
     assert response.json()["price"] == float(update_data.price)
 
 
-def test_authenticated_user_can_delete_category(
+def test_superuser_can_delete_product(
     sync_client: TestClient,
-    auth_headers: dict[str, str],
+    superuser_auth_headers: dict[str, str],
     db_products: list[CategoryOutputSchema],
 ):
-    response = sync_client.delete(f"products/{db_products[0].id}", headers=auth_headers)
+    response = sync_client.delete(
+        f"products/{db_products[0].id}", headers=superuser_auth_headers
+    )
     assert response.status_code == status.HTTP_204_NO_CONTENT
 
 
-def test_authenticated_user_can_delete_all_products(
+def test_superuser_can_delete_all_products(
     sync_client: TestClient,
-    auth_headers: dict[str, str],
+    superuser_auth_headers: dict[str, str],
     db_products: list[ProductOutputSchema],
 ):
-    response = sync_client.delete("products/", headers=auth_headers)
+    response = sync_client.delete("products/", headers=superuser_auth_headers)
     assert response.status_code == status.HTTP_204_NO_CONTENT
 
-    response = sync_client.get("products/", headers=auth_headers)
+    response = sync_client.get("products/", headers=superuser_auth_headers)
+    assert response.status_code == status.HTTP_200_OK
     assert response.json()["total"] == 0
 
 
+def test_authenticated_user_cannot_get_products(
+    sync_client: TestClient, auth_headers: dict[str, str]
+):
+    response = sync_client.get("products/", headers=auth_headers)
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
```

Comment:
authenticated user should be able to get products (all or single one), maybe even unauthenticated user should be able to do that because it is only about GET action. IRL we are able to scroll through the product list even if we are not logged in